uninstall.sh

```diff
@@ -0,0 +1,350 @@
+#!/bin/bash
+set -u
+shopt -s extglob
+
+# Set up temp dir
+tmpdir=/tmp/uninstall.$$
+mkdir -p "$tmpdir" || Kernel_abort "Unable to create temp dir '$tmpdir'"
+trap '
+  rm -fr "$tmpdir"
+  # Invalidate sudo timestamp before exiting
+  /usr/bin/sudo -k
+' exit
+
+### Shell equivalents to standard Ruby methods
+
+Kernel_abort() {
+  [[ $1 ]] && echo "$1" 2>/dev/null
+  exit 1
+}
+
+String_strip() {
+  local s
+  for s in "$@"; do
+    s=${s## }
+    echo "${s%% }"
+  done
+}
+
+Dir_children() {
+  local p
+  for p in "$@"; do
+    [[ -d $p ]] || continue
+    find "$p" -maxdepth 1 | grep -v "^${p}$"
+  done
+}
+
+###
+
+# Default options
+opt_force=""
+opt_quiet=""
+opt_dry_run=""
+opt_skip_cache_and_logs=""
+
+# global status to indicate whether there is anything wrong.
+failed=false
+
+un=$(uname)
+case "$un" in
+  Linux)
+    ostype=linux
+    homebrew_prefix_default=/home/linuxbrew/.linuxbrew
+  ;;
+  Darwin)
+    ostype=macos
+    homebrew_prefix_default=/usr/local
+    # Stock macOS doesn't have a realpath, so we make do with builtin Python
+    realpath() {
+      /usr/bin/python -c 'import os,sys;print(os.path.realpath(sys.argv[1]))' "$1"
+    }
+  ;;
+  *)
+    Kernel_abort "Unsupported system type '$un'"
+  ;;
+esac
+
+# string formatters
+if [[ -t 1 ]]; then
+  Tty_escape() { printf "\033[%sm" "$1"; }
+else
+  Tty_escape() { :; }
+fi
+Tty_mkbold() { Tty_escape "1;${1:-39}"; }
+Tty_blue=$(Tty_mkbold 34)
+Tty_red=$(Tty_mkbold 31)
+Tty_bold=$(Tty_mkbold 39)
+Tty_reset=$(Tty_escape 0)
+
+Array_shell_s() {
+  local result=("$1"); shift
+  local s
+  for s in "$@"; do
+    result+=("${s// /\\ }")
+  done
+  echo -n "${result[*]}"
+}
```

Comment:
Why is this implementation different from `shell_join` in installer.sh?

Reply:
Mostly because it didn't click in my mind that they were actually the same function. Will replace with `shell_join`.

Reply:
Along with any other "twinned" functions.

Reply:
Agreed that consistency with install.sh would be good for now.